src/Datadog.Trace.ClrProfiler.Managed/Emit/ObjectExtensions.cs

```diff
@@ -49,6 +49,41 @@ internal static class ObjectExtensions
             return false;
         }
 
+        /// <summary>
+        /// Tries to call an instance method with the specified name, two parameters, and no return value.
+        /// </summary>
+        /// <typeparam name="TArg1">The type of the method's first parameter.</typeparam>
+        /// <typeparam name="TArg2">The type of the method's second parameter.</typeparam>
+        /// <param name="source">The object to call the method on.</param>
+        /// <param name="methodName">The name of the method to call.</param>
+        /// <param name="arg1">The value to pass as the method's first argument.</param>
+        /// <param name="arg2">The value to pass as the method's second argument.</param>
+        /// <returns><c>true</c> if the method was found, <c>false</c> otherwise.</returns>
+        public static bool TryCallVoidMethod<TArg1, TArg2>(this object source, string methodName, TArg1 arg1, TArg2 arg2)
+        {
+            var type = source.GetType();
+            var paramType1 = typeof(TArg1);
+            var paramType2 = typeof(TArg2);
+
+            object cachedItem = Cache.GetOrAdd(
+                $"{type.AssemblyQualifiedName}.{methodName}.{paramType1.AssemblyQualifiedName}.{paramType2.AssemblyQualifiedName}",
```

Comment:
This will cause a merge conflict with my review. Since your is more important, I'm going to wait before merging mine 👍